.github/workflows/cross-version-tests.yml

```diff
@@ -75,14 +75,13 @@ jobs:
       - id: set-matrix
         name: Set matrix
         run: |
-          if [ ! "$(git branch --show-current)" == "master" ]; then
-            git fetch origin master:master
-          fi
-
           EVENT_NAME="${{ github.event_name }}"
           if [ "$EVENT_NAME" = "pull_request" ]; then
-            REF_VERSIONS_YAML="https://raw.githubusercontent.com/mlflow/mlflow/master/mlflow/ml-package-versions.yml"
-            CHANGED_FILES="$(git diff --name-only master..HEAD)"
+            REPO="${{ github.repository }}"
+            PR_NUMBER="${{ github.event.pull_request.number }}"
+            REF_VERSIONS_YAML="https://raw.githubusercontent.com/$REPO/master/mlflow/ml-package-versions.yml"
+            API_URL="https://api.github.com/repos/$REPO/pulls/$PR_NUMBER/files?per_page=100"
```

Comment:
Does this imply that this method will fail if a PR changed more than 100 files? (I know this is extremely unlikely, but just want to call out)

Reply:
“this method will fail” means that changes in the 101+ files will not be used to trigger the cross-version test, right?

Reply:
No, they won't be used. Let me create a python script that fetches all changed files in the PR.

Reply:
Added a python script to fetch all changed files!

Reply:
Thanks for enhancing the solution!